core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -94,7 +94,7 @@ private[spark] object SignalUtils extends Logging {
 
       // run all actions, escalate to parent handler if no action catches the signal
       // (i.e. all actions return false)
-      val escalate = actions.asScala.forall { action => !action() }
+      val escalate = actions.asScala.map(action => action()).forall(_ == false)
```

Comment:
we should probably leave a comment here saying why we are doing map and then forall, to make sure nobody would come in and "optimize" the map.forall into just a forall.